File: DTSparkCore/src/main/java/com/kevin/java/transformations/Parallelize_RDD.java
package com.kevin.java.transformations;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.VoidFunction;

import java.util.Arrays;
import java.util.List;

/**
 * @author kevin
 * @version 1.0
 * @description     parallelize将list数据转为RDD
 * @createDate 2018/12/28
 */
public class Parallelize_RDD {

    public static void main(String[] args) {

        // 1.创建SparkConf，设置作业名称和模式
        SparkConf conf = new SparkConf().setAppName("Parallelize_RDD").setMaster("local");

        // 2.基于Sparkconf对象创建一个SparkContext上下文，它是通往集群的唯一通道，且在创建时会创建任务调度器
        JavaSparkContext sc = new JavaSparkContext(conf);

        List<String> list = Arrays.asList("Java", "scala", "Python", "Java");

        // 3.去重
        JavaRDD<String> parallelize = sc.parallelize(list);

        // 4.遍历
        parallelize.foreach(new VoidFunction<String>() {
            @Override
            public void call(String s) throws Exception {
                System.out.println(s);
            }
        });

        // 5.关闭
        sc.close();

    }
}

File: dt-spark-core/src/main/scala/com/kevin/scala/transformations/Parallelize_RDD.scala
package com.kevin.scala.transformations

import org.apache.spark.{SparkConf, SparkContext}

/**
  * parallelize将list数据转为RDD
  */
object Parallelize_RDD {

  def main(args: Array[String]): Unit = {
    // 1.创建SparkConf
    val conf = new SparkConf().setAppName("Parallelize_RDD").setMaster("local")
    // 2.创建SparkContext
    val sc = new SparkContext(conf)
    // 3.创建集合数据
    val list = List("Java", "scala", "Python", "Java")
    // 4.parallelize将集合转成rdd,foreach遍历输出
    sc.parallelize(list).foreach(println(_))
    // 5.关闭sc
    sc.stop()

  }

}
